### PR TITLE
Private testing

### PR DESCRIPTION
**Pull request by abhinavkadiyala**
Abhinav's version of private unit testing solution.

### CODE REVIEW

File: .idea/misc.xml [hunk 1]
```diff
@@ -37,7 +37,7 @@
     <ConfirmationsSetting value="0" id="Add" />
     <ConfirmationsSetting value="0" id="Remove" />
   </component>
-  <component name="ProjectRootManager" version="2" languageLevel="JDK_1_8" default="true" assert-keyword="true" jdk-15="true" project-jdk-name="1.8" project-jdk-type="JavaSDK">
```
Comment (Parthib) [inline .idea/misc.xml:40]: Downgrading JDK is bad

File: app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java [hunk 2]
```diff
@@ -7,7 +7,7 @@
 
 import com.android.volley.toolbox.ImageLoader;
 
-class LruBitmapCache extends LruCache<String, Bitmap>
```
Comment (Parthib) [inline app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java:10]: Shouldn't this be private?

File: app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
+package com.example.samplegallery;
+
+/**
+ * Created by Kadiyala on 1/16/17.
+ */
+
+import android.content.Context;
```
Comment (Parthib) [inline app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java:7]: I don't think imports are supposed to be spaced apart

File: app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+import com.example.samplegallery.Utilities.*;
+
+import org.junit.Test;
+
```
Comment (Parthib) [inline app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java]: Too many new lines

File: app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+public class LruBitmapCacheTest {
+
+    @Test
+    public void test()
+    {
+        Context context;
+
+        LruBitmapCache.getCacheSize(context);
+
```
Comment (Parthib) [inline app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java]: Too many new lines

File: build.gradle [hunk 6]
```diff
@@ -5,7 +5,7 @@ buildscript {
         jcenter()
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:2.2.1'
```
Comment (Parthib) [inline build.gradle:8]: Changes in gradle build version might be bad - ask Stan

File: app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java [hunk 7]
```diff
@@ -36,7 +36,7 @@ public void putBitmap(String url, Bitmap bitmap) {
     }
 
     // Make cache of size such that it may hold ~3 screens worth of images
-    public static int getCacheSize(Context ctx) {
+    static int getCacheSize(Context ctx) {
```
Comment (Parthib) [inline app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java]: This should be private?

File: app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java [hunk 8]
```diff
@@ -7,7 +7,7 @@
```
Comment (lasyasreepada) [inline app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java:7]: Unnecessary white space?

File: app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java (hunk 2 above)
Comment (lasyasreepada) [reply app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java:10]: agree - should be private not package private

File: app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java [hunk 9]
```diff
@@ -0,0 +1,89 @@
...
+import android.util.DisplayMetrics;
+
+import com.example.samplegallery.Utilities.*;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.runners.MockitoJUnitRunner;
+import org.mockito.*;
+
+import static org.hamcrest.MatcherAssert.assertThat;
```
Comment (lasyasreepada) [inline app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java:17]: I would put all your static imports first, then organize the non static imports alphabetically. Usually only space between static and non static imports, but I'm not sure on this?

File: app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java [hunk 10]
```diff
@@ -0,0 +1,89 @@
...
+
+        int cachesize = 0;
+        try {
+            cachesize = (int) getCacheSize.invoke(LruBitmapCache.class, mockContext);
+        } catch (IllegalAccessException e) {
+            e.printStackTrace();
+        } catch (InvocationTargetException e) {
+            e.printStackTrace();
+        }
+
```
Comment (lasyasreepada) [inline app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java:84]: Too much whitespace?

File: app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java [hunk 11]
```diff
@@ -0,0 +1,89 @@
...
+        mockedDisplaymetrics.widthPixels = 4;
+        mockedDisplaymetrics.heightPixels = 5;
+
+        when(mockContext.getResources().getDisplayMetrics()).thenReturn(mockedDisplaymetrics);
+
+        Method getCacheSize = null;
+        try {
+            getCacheSize = LruBitmapCache.class.getDeclaredMethod("getCacheSize", Context.class);
+            getCacheSize.setAccessible(true);
+        } catch (NoSuchMethodException e) {
```
Comment (lasyasreepada) [inline app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java:70]: Are all these catch blocks necessary? maybe just include a throws clause

File: app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java (hunk 8 above)
Comment (TaishoJapan) [inline app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java]: * 3?

Comment (TaishoJapan) [inline app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java]: in place of 4

File: app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java (hunk 2 above)
Comment (abhinavkadiyala) [inline app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java]: thanks I fixed it

File: app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java (hunk 7 above)
Comment (abhinavkadiyala) [reply app/src/main/java/com/example/samplegallery/Utilities/LruBitmapCache.java]: thanks fixed from package private to private

File: app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java [hunk 12]
```diff
@@ -5,23 +5,83 @@
...
 
+        int cachesize = LruBitmapCache.getCacheSize(mockContext);
+
+
+        assertThat(cachesize,is(240));
 
     }
 
-}
+    // For private getCacheSize method
```
Comment (Parthib) [inline app/src/test/java/com/example/samplegallery/LruBitmapCacheTest.java]: Might consider just making a  new branch for each

**Pull request closed without merging**